actors/verifreg/src/emit.rs

```diff
@@ -13,87 +15,221 @@ use fvm_shared::ActorID;
 pub fn verifier_balance(
     rt: &impl Runtime,
     verifier: ActorID,
+    prev_balance: &DataCap,
     new_balance: &DataCap,
+    client: Option<ActorID>,
 ) -> Result<(), ActorError> {
     rt.emit_event(
         &EventBuilder::new()
             .typ("verifier-balance")
             .field_indexed("verifier", &verifier)
+            .field_indexed("client", &client)
+            .field("prev-balance,", &BigIntSer(prev_balance))
             .field("balance", &BigIntSer(new_balance))
             .build()?,
     )
 }
 
 /// Indicates a new allocation has been made.
+#[allow(clippy::too_many_arguments)]
 pub fn allocation(
     rt: &impl Runtime,
     id: AllocationID,
     client: ActorID,
     provider: ActorID,
+    piece_cid: &Cid,
+    piece_size: &u64,
+    term_min: &ChainEpoch,
+    term_max: &ChainEpoch,
+    expiration: &ChainEpoch,
```

Comment:
This is a lot of parameters, many of the same type, and hard to verify that the right things are passed in the right place at the call sites. At this point I think you should just take an `&Allocation` as one parameter. From a quick skim, this will work at the call sites. Ditto for claims.

Reply:
Fixed this everywhere.